modules/shared/src/main/scala/org/tessellation/ext/collection/IterableUtils.scala

```diff
@@ -0,0 +1,6 @@
+package org.tessellation.ext.collection
+
+object IterableUtils {
+  def pickMajority[A](as: Iterable[A]): Option[A] =
```

Comment:
I think it would be best to extract the majority function from `ConsensusStateUpdater` and use that one everywhere.

Reply:
> I think it would be best to extract the pick majority function from ConsensusStateUpdater and use that one everywhere.

The implementation I replaced it with is equivalent and I believe more efficient.

Reply:
@sdavidp then, it should be used in the ConsensusStateUodater as well 🤔 to have a single implementation only and be consistent across various processes

Reply:
@marcinwadon Yes, I did that.

Reply:
Great. Didn't check the full PR yet. Btw. it's my personal opinion but I really don't like "utils" name as it's a container for everything. I'd rather keep the pattern used internally in cats (trait "...Ops" and then object extending the trait) or an implicit class on Iterable to extend its functionality.

Reply:
One thing about the implementation in `ConsensusStateUpdater` is that it's based on the supplied `Order` for the element that we pick majority on. And the advantage of it is that you get deterministic result no matter where you run it for the case where the same elements have the same number of appearances. If that's the case then the order for `A` decides which element is picked as majority - we need that property to get deterministic results on every machine running this code. At first I also thought that the order is used for grouping the elements, but I think it's not the case unfortunately.

Reply:
Is this conversation resolved?

Reply:
As per the discussion in the meeting, this test passes
```
pureTest("pickMajority") {
  import org.tessellation.ext.collection.FoldableOps._
  val majority = pickMajority(List(2, 4, 0, 1, 3))
  expect.all(
    majority.contains(4),
    !majority.contains(0)
  )
}
```

Reply:
Resolving; issue has been addressed